src/shell/interpreter.zig

```diff
@@ -1125,24 +1107,15 @@ pub const Interpreter = struct {
                 .stdin = .{
                     .fd = stdin_reader,
                 },
-                .stdout = .{
-                    .fd = .{
-                        .writer = stdout_writer,
-                    },
-                },
-                .stderr = .{
-                    .fd = .{
-                        .writer = stderr_writer,
-                    },
-                },
+                // By default stdout/stderr should be an IOWriter writing to a dup'ed stdout/stderr
+                // But if the user later calls `.setQuiet(true)` then all those syscalls/initialization was pointless work
+                // So we cheaply initialize them now as `.pipe`
+                // When `Interpreter.run()` is called, we check if `this.flags.quiet == false`, if so then we then properly initialize the IOWriter
+                .stdout = .pipe,
```

Comment:
Does this work on windows? on windows we need to create the uv.Pipe ahead of time IIRC